### PR TITLE
Add Microsoft Teams webhook support for notifications

### LINKED ISSUE #4071: [DOC] Notifications - Support Microsoft Teams webhooks
**What do you want to do?**
 
- [x] Request a change to existing documentation
- [ ] Add new documentation
- [ ] Report a technical problem with the documentation
- [ ] Other 

**Tell us about your request.** Provide a summary of the request and all versions that are affected.
 Change existing documentation, add content about microsoft teams webhook. User can add microsoft teams webhook as notification channel, and send notifications to this channel.
 It will affect the versions after 2.8(including 2.8).

**What other resources are available?** Provide links to related issues, POCs, steps for testing, etc.
[opensearch-project/notifications#585](https://github.com/opensearch-project/notifications/issues/585)

### PR DESCRIPTION
**Pull request by kolchfa-aws**
Fixes #4071 

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by zhichao-aws — APPROVED** [review]
Looks good to me.

File: _observing-your-data/notifications/api.md [hunk 1]
```diff
@@ -121,6 +122,7 @@ email.recipient_list | The channel's recipient list.
 email_group.recipient_list | The channel's list of email recipient groups.
 smtp_account.method | The email encryption method.
 slack.url	| The Slack channel's URL.
+microsoft_teams.url | The Microsoft Teams channel's URL.
```
Comment (natebower) [inline _observing-your-data/notifications/api.md:125]:
```suggestion
microsoft_teams.url | The Microsoft Teams channel URL.
```

File: _observing-your-data/notifications/index.md [hunk 2]
```diff
@@ -23,7 +23,7 @@ You can use either OpenSearch Dashboards or the REST API to configure notificati
...
-2. In the **Configurations** section, select the channel type and enter the necessary information for each type. For more information about configuring a channel that uses Amazon SNS or email, refer to the sections below. If you want to use Amazon Chime or Slack, you need to specify the webhook URL. For more information about using webhooks, see the documentation for [Slack](https://api.slack.com/messaging/webhooks) and [Amazon Chime](https://docs.aws.amazon.com/chime/latest/ug/webhooks.html).
+2. In the **Configurations** section, select the channel type and enter the necessary information for each type. For more information about configuring a channel that uses Amazon SNS or email, refer to the sections below. If you want to use Amazon Chime or Slack, you need to specify the webhook URL. For more information about using webhooks, see the documentation for [Slack](https://api.slack.com/messaging/webhooks), [Microsoft Teams](https://learn.microsoft.com/en-us/microsoftteams/platform/webhooks-and-connectors/what-are-webhooks-and-connectors), and [Amazon Chime](https://docs.aws.amazon.com/chime/latest/ug/webhooks.html).
```
Comment (natebower) [inline _observing-your-data/notifications/index.md:26]:
```suggestion
2. In the **Configurations** section, select the channel type and enter the necessary information for each type. For more information about configuring a channel that uses Amazon SNS or email, refer to the sections below. If you want to use Amazon Chime or Slack, you need to specify the webhook URL. For more information about using webhooks, see the documentation for [Slack](https://api.slack.com/messaging/webhooks), [Microsoft Teams](https://learn.microsoft.com/en-us/microsoftteams/platform/webhooks-and-connectors/what-are-webhooks-and-connectors), or [Amazon Chime](https://docs.aws.amazon.com/chime/latest/ug/webhooks.html).
```

**Review by natebower — APPROVED** [review]
@kolchfa-aws Two minor changes, though it looks like the first one may be applicable to the entire list. Thanks!

**Pull request merged**